README.md

```diff
@@ -16,86 +16,51 @@ If you want to learn more about modules, [this page in the Seesaw guide](https:/
-let mut neokeys = NeoKey1x4::new_with_default_addr(seesaw.acquire_driver())
-    .init()
-    .expect("Failed to start NeoKey1x4");
+let seesaw = SeesawDriver::new(delay, i2c);
+let neokeys = NeoKey1x4::new_with_default_addr(seesaw).init().unwrap();
 ```
 
-## `std` (multi-threaded)
+## Multiple Devices
 
-> This requires turning on the `std` feature flag.
+### `no_std`
 
-For multi-threaded purposes, use the `SeesawStdMutex` typed struct, which wraps the bus in a std `Mutex`.
+For multiple devices, use third-party libraries like [`embedded-hal-bus`](https://crates.io/crates/embedded-hal-bus) and [`embassy-time`](https://crates.io/crates/embassy-time) to facilitate bus and delay sharing.
 
-Example usage of using multi-threaded `Seesaw` in a `std` context, running on an ESP32-S3:
+[Complete example here.](https://github.com/alexeden/adafruit-seesaw/blob/main/examples/embedded_hal_rotary_encoder_test.rs)
 
 ```rs
-use adafruit_seesaw::{devices::RotaryEncoder, prelude::*, SeesawStdMutex};
-use esp_idf_hal::{
-    self,
-    delay::Delay,
-    gpio::PinDriver,
-    i2c::{I2cConfig, I2cDriver},
-    peripherals::Peripherals,
-    prelude::*,
-};
-use std::time::Duration;
-
-fn main() -> Result<(), anyhow::Error> {
-    esp_idf_hal::sys::link_patches();
-    esp_idf_svc::log::EspLogger::initialize_default();
-
-    // System
-    let peripherals = Peripherals::take().unwrap();
-    let mut i2c_power = PinDriver::output(peripherals.pins.gpio7).unwrap();
-    i2c_power.set_low()?;
-    std::thread::sleep(Duration::from_millis(333));
-
-    // I2C
-    let (sda, scl) = (peripherals.pins.gpio3, peripherals.pins.gpio4);
-    let config = I2cConfig::new().baudrate(400.kHz().into());
-    let i2c = I2cDriver::new(peripherals.i2c0, sda, scl, &config)?;
-    i2c_power.set_high()?;
-    std::thread::sleep(Duration::from_millis(333));
-
-    let seesaw: &'static _ = {
-        use once_cell::sync::OnceCell;
-        static MANAGER: OnceCell<SeesawStdMutex<(Delay, I2cDriver<'_>)>> =
-            OnceCell::new();
-
-        match MANAGER.set(SeesawStdMutex::new(Delay::new_default(), i2c)) {
-            Ok(_) => MANAGER.get(),
-            Err(_) => None,
-        }
-    }
-    .unwrap();
+// Setup on an STM32F405
+use embassy_time::Delay;
+use embedded_hal_bus::i2c::RefCellDevice;
 
-    let _encoder =
-        RotaryEncoder::new_with_default_addr(seesaw.acquire_driver())
-            .init()
-            .expect("Failed to start rotary encoder.");
+let clocks = dp.RCC.constrain().cfgr.freeze();
+let i2c = RefCell::new(I2c::new(dp.I2C1, (scl, sda), 400.kHz(), &clocks));
 
-    loop {
-        // Do stuff with rotary encoder
-    }
-}
+let encoder_driver_1 = SeesawDriver::new(Delay, RefCellDevice::new(&i2c));
+let encoder_1 = RotaryEncoder::new(0x00, encoder_driver_1).init().unwrap();
+
+let encoder_driver_2 = SeesawDriver::new(Delay, RefCellDevice::new(&i2c));
+let encoder_2 = RotaryEncoder::new(0x01, encoder_driver_2).init().unwrap();
 ```
 
+### `std`
```

Comment:
i'm not sure if it makes sense that you explicitly document the std/no_std differentiation here - you might want to leave this up to `embedded-hal-bus` (you anyway already link there). because their docs (can) document all the nuances much better and there's nothing specific to be gained here for seesaw devices?